lib/oli_web/live/workspaces/student.ex

```diff
@@ -429,8 +416,38 @@ defmodule OliWeb.Workspaces.Student do
     }
   end
 
-  defp user_is_only_a_student?(%{can_create_sections: true}, _sections), do: false
+  defp user_is_only_a_student?(%{can_create_sections: true}), do: false
+
+  defp user_is_only_a_student?(%{id: user_id}) do
+    !Repo.exists?(
+      from e in Enrollment,
+        left_join: ecr in EnrollmentContextRole,
+        on: e.id == ecr.enrollment_id,
+        left_join: upr in "users_platform_roles",
+        on: e.user_id == upr.user_id,
+        where: e.user_id == ^user_id,
+        where:
+          ecr.context_role_id not in ^@context_student_roles_ids or
+            upr.platform_role_id not in ^@platform_student_roles_ids
+    )
+  end
 
-  defp user_is_only_a_student?(_user, sections),
-    do: Enum.all?(sections, &(&1.user_role == "student"))
+  defp sections_where_user_is_student(user_id) do
+    Repo.all(
+      from s in Section,
+        join: e in Enrollment,
+        on: s.id == e.section_id,
+        left_join: ecr in EnrollmentContextRole,
+        on: e.id == ecr.enrollment_id,
+        left_join: upr in "users_platform_roles",
+        on: e.user_id == upr.user_id,
+        where: e.user_id == ^user_id,
+        where: s.open_and_free == true,
+        where: s.status == :active,
+        where:
+          ecr.context_role_id in ^@context_student_roles_ids or
+            upr.platform_role_id in ^@platform_student_roles_ids,
+        select: s
+    )
+  end
```

Comment:
This query needs a context and some tests.

Reply:
👍 added tests in `test/oli/delivery/sections_test.exs`